imap_processing/hit/l1a/hit_l1a.py

```diff
@@ -0,0 +1,208 @@
+"""Decommutate HIT CCSDS data and create L1a data products."""
+
+import logging
+from collections import defaultdict
+from dataclasses import fields
+from enum import IntEnum
+
+import numpy as np
+import xarray as xr
+
+from imap_processing import decom, imap_module_directory, utils
+from imap_processing.cdf.global_attrs import ConstantCoordinates
+from imap_processing.cdf.utils import write_cdf
+from imap_processing.hit import hit_cdf_attrs
+from imap_processing.hit.l0.data_classes.housekeeping import Housekeeping
+
+logger = logging.getLogger(__name__)
+
+
+class HitAPID(IntEnum):
+    """
+    HIT APID Mappings.
+
+    Attributes
+    ----------
+    HIT_AUT : int
+        Autonomy
+    HIT_HSKP: int
+        Housekeeping
+    HIT_SCIENCE : int
+        Science
+    HIT_IALRT : int
+        I-ALiRT
+    HIT_MEMDUMP : int
+        Memory dump
+    """
+
+    HIT_AUT = 1250  # Autonomy
+    HIT_HSKP = 1251  # Housekeeping
+    HIT_SCIENCE = 1252  # Science
+    HIT_IALRT = 1253  # I-ALiRT
+    HIT_MEMDUMP = 1255  # Memory dump
+
+
+def hit_l1a_data(packet_file: Path | str):
+    """
+    Process HIT L0 data into L1A data products.
+
+    Parameters
+    ----------
+    packet_file : str
+        Path to the CCSDS data packet file.
+
+    Returns
+    -------
+    dict
+        List of file paths to CDF data product files
+    """
+    # Decom, sort, and group packets by apid
+    packets = decom_packets(packet_file)
+    sorted_packets = sorted(packets, key=lambda x: x.data["SHCOARSE"].derived_value)
+    grouped_data = group_data(sorted_packets)
+
+    # Create datasets
+    # TODO define keys to skip for each apid. Currently just have
+    #  a list for housekeeping.
+    skip_keys = [
+        "shcoarse",
+        "ground_sw_version",
+        "packet_file_name",
+        "ccsds_header",
+        "leak_i_raw",
+    ]
+    datasets = create_datasets(grouped_data, skip_keys)
```

Comment:
Also, This function may be helpful here. https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/e1234fe0f69ab76a3d3484114abb8f200b2e1989/imap_processing/utils.py#L131

Reply:
SWAPI has example of how it's using above functions, https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/e1234fe0f69ab76a3d3484114abb8f200b2e1989/imap_processing/swapi/l1/swapi_l1.py#L587

Reply:
I looked at those examples when I was working on this code and they were super helpful! I had to create a custom create_dataset function because our packets are data classes and the global function expects a list of space packet parser packets. Maybe there's a better way to handle this difference in data format?

Reply:
I see. Yeah, we can revisit that in future PR.